Sources/SwiftParser/Expressions.swift

```diff
@@ -1187,57 +1148,61 @@ extension Parser {
 }
 
 extension Parser {
-  /// Parse an identifier as an expression.
+  /// Parse a macro expansion as an expression.
+  ///
   ///
   /// Grammar
   /// =======
   ///
-  ///     playground-literal → '#colorLiteral' '(' red ':' expression , green ':' expression , blue ':' expression , alpha ':' expression )
-  ///     playground-literal → '#fileLiteral' '(' resourceName ':' expression ')'
-  ///     playground-literal → '#imageLiteral' '(' resourceName ':' expression ')'
+  /// macro-expansion-expression → '#' identifier expr-call-suffix?
   @_spi(RawSyntax)
-  public mutating func parseObjectLiteralExpression() -> RawObjectLiteralExprSyntax {
+  public mutating func parseMacroExpansionExpr(
+    pattern: PatternContext,
+    flavor: ExprFlavor
+  ) -> RawMacroExpansionExprSyntax {
     let poundKeyword = self.consumeAnyToken()
-    let (unexpectedBeforeLeftParen, leftParen) = self.expect(.leftParen)
-    let arguments = self.parseArgumentListElements(pattern: .none)
-    let (unexpectedBeforeRightParen, rightParen) = self.expect(.rightParen)
-    return RawObjectLiteralExprSyntax(
-      identifier: poundKeyword,
-      unexpectedBeforeLeftParen,
+    let (unexpectedBeforeMacro, macro) = self.expectIdentifier()
+
+    // Parse the optional parenthesized argument list.
+    let leftParen = self.consume(if: .leftParen, where: { !$0.isAtStartOfLine })
```

Comment:
AFAICT `parseMacroExpansionExpr` and `parseMacroExpansionDecl` seem pretty similar to me. Do you think we could unify their parsing?